internal/pkg/apis/cosigned/clusterimagepolicy_types.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusterimagepolicy
+
+import (
+	"crypto/ecdsa"
+
+	"github.com/sigstore/cosign/pkg/apis/cosigned/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ClusterImagePolicy defines the images that go through verification
+// and the authorities used for verification.
+// This is the internal representation of the external v1alpha1.ClusterImagePolicy.
+// KeyRef does not store secretRefs in internal representation.
+// KeyRef does store parsed publicKeys from Data in internal representation.
+
+type ClusterImagePolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec holds the desired state of the ClusterImagePolicy (from the client).
+	Spec InternalClusterImagePolicySpec `json:"spec"`
+}
+
+type InternalClusterImagePolicySpec struct {
+	Images      []v1alpha1.ImagePattern `json:"images"`
+	Authorities []Authority             `json:"authorities"`
+}
+
+type Authority struct {
+	// +optional
+	Key *KeyRef `json:"key,omitempty"`
+	// +optional
+	Keyless *v1alpha1.KeylessRef `json:"keyless,omitempty"`
+	// +optional
+	Sources []v1alpha1.Source `json:"source,omitempty"`
+	// +optional
+	CTLog *v1alpha1.TLog `json:"ctlog,omitempty"`
+}
+
+// This references a public verification key stored in
+// a secret in the cosign-system namespace.
+type KeyRef struct {
+	// Data contains the inline public key
+	// +optional
+	Data string `json:"data,omitempty"`
```

Comment:
If we don't need it, let's remove it.

Reply:
Discussed an approach with @hectorj2f , going to try to make that change.
- dont use unmarshal/marshal
- have a Convert function that does the conversion between external to internal CIP

Reply:
I'm fine with actually checking this in, and improving the conversion function in the follow on PRs, since:
 * That's all going to be internal to the reconciler
 * API won't change (Data will be dropped, is not used).
So, up to you if you want to add to this PR, or just merge this as is and then work on the conversion in a follow up?